app/components/DataInput/index.js

```diff
@@ -47,21 +46,16 @@ class DataInput extends AppComponent {
     ];
   }
 
-  componentWillMount() {
-    this.setState({ rawData: this.props.rawData });
-  }
-
   componentWillReceiveProps(nextProps) {
     if (this.props.rawData !== nextProps.rawData) {
       this.setState({ rawData: nextProps.rawData });
     }
   }
 
   _submitData(data) {
-    // setState is async so we apply trim() twice
-    this.setState({ rawData: data.trim() });
-    this.props.dispatch(
-      actionTrigger(RECEIVE_RAW_DATA, data.trim())
+    const rawData = data.trim();
+    this.setState({ rawData }, () =>
```

Comment:
I do not like that we are updating the local state and also the redux state...

Is there a reason we need to do this and not just do it in the redux state?